### PR TITLE
Use hc-spin cli 

### PR DESCRIPTION
**Pull request by matthme**
Switches to using the hc-spin CLI instead of hc launch by default since it provides better developer tools.

### CODE REVIEW

**Review by ThetaSinner — APPROVED** [review]

**Review by c12i — COMMENTED** [review]
Apart from the `--host` fix for `No index.html file found at http://127.0.0.1:8888/index.html` on the `ui/package.json`, everything else works as expected

**Comment by c12i** [comment]
@ThetaSinner I thin this change should only be backported to `develop-0.2`

Since we already have #200 targeting `develop-01`, which I think we should still keep open since the launch scripts are pretty different between `0.1` and `0.2`

**Comment by ThetaSinner** [comment]
That works. Ideally the only difference would be the version of `hc-spin` we're pulling in and yes that would need adjusting on a back-port but that's expected.

If you're happy that the changes are equivalent between the two PRs then that's fine with me. 

For context, the merge to develop then back-port strategy is partly there to ensure we don't lose changes. If bugfixes are submitted directly to the 0.1 and 0.2 branches then it's easy to lose them. That's not a problem here, but in general it's easier to keep track of if general enhancements have a singe place that they are supposed to be made

File: src/versions.rs [hunk 1]
```diff
@@ -3,7 +3,11 @@ pub fn tryorama_version() -> String {
 }
 
 pub fn holochain_client_version() -> String {
-    String::from("^0.16.5")
+    String::from("^0.16.7")
```
Comment (c12i) [inline src/versions.rs:6]: @matthme is this upgrade needed?

`develop` is currently on `^0.17.0-dev.3`

**Comment by c12i** [comment]
@matthme I haven't tested `hc-spin` with `0.3` yet, and I am not sure whether it's compatible with it too. I think to be on the safe side, we can change the target branch of this PR to `develop-0.2` instead of `develop`. Since `develop` has now been upgraded to be compatible `0.3` (#191) since this PR was last reviewed

cc @ThetaSinner

**Comment by c12i** [comment]
I just noticed that a `0.3` compatible version of client was shipped with the vue.js `cell_id` proxy object fix

I have opened #205 to use the latest version

**Comment by matthme** [comment]
> I haven't tested hc-spin with 0.3 yet, and I am not sure whether it's compatible with it too

Yes, I haven't tested it with 0.3 either and I suspect that I will need to make a dedicated hc-spin version `0.300.x` for that. Changing the target branch to develop-0.2 is fine with me. And I will test hc-spin with 0.3 now that scaffolding is available for it.

**Comment by c12i** [comment]
@matthme, modifying the base branch has introduced commits not present in `develop-0.2` from `develop`, leading to merge conflicts.I recommend reopening this PR against `develop-0.2` to address this issue. What are your thoughts on this approach?

**Comment by c12i** [comment]
Alternatively, I can close this PR and backport #208 into the respective maintenance branches using the correct holochain client version for each

**Pull request closed without merging**